torchdata/datapipes/iter/load/huggingface.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+import posixpath
+
+from typing import Any, Callable, Dict, Iterator, List, Optional, Tuple, Union
+
+from torch.utils.data.datapipes.utils.common import match_masks
+
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+from torchdata.datapipes.utils import StreamWrapper
+
+try:
+    import datasets
+except ImportError:
+    datasets = None
+
+def _get_response_from_huggingface_hub(dataset, split, revision, data_files) -> Tuple[Any, StreamWrapper]:
+    dataset = datasets.load_dataset(dataset, split, revision, data_files)
```

Comment:
n00b question:
1. Their API has an option `streaming=True` for an IterableDataset, do we want that?
2. How do people typically do something like `for batch in dataset`? In that case, do we want `yield from` instead in the DataPipe?

Reply:
that's a good callout, I think using `streaming=True` by default is a good idea and users should be able to overwrite that behavior

A `yield` also sounds good to me, I was using this `    return dataset[0], StreamWrapper(dataset)` because that's what the other examples seemed to be using